### ISSUE TITLE
llama.cpp cannot be compiled in dockerfile (invalid instruction set errors)

### ISSUE THREAD

**Issue by nsarrazin** [nsarrazin/serge#82]
This is the followup ticket to #66. 

We can't use a compiled binary of llama.cpp in our dockerfile because the compilation step checks which instruction set is available on the cpu.

Compiling it on a github worker results in `invalid instruction set` errors being thrown at runtime, since the CPU of the worker might not be the same as the user.

This is linked to the following discussion in the llama.cpp repo: https://github.com/ggerganov/llama.cpp/discussions/535

We can't really wait on them as this might be a while. 

Not sure how to tackle this on our end. I think the easiest option would be to just compile llama.cpp at runtime, maybe on server startup in the same step where we check models .bin for compatibility.  This has plenty of drawbacks but seems like the only alternative so far, to me at least. Any thoughts ?

**Comment by FenarkSEC** [comment]
How much ressource and time would it take to compile llama.cpp at runtime ? Wouldn't the ```invalid instruction set``` be moved to the end-user instead if we go that way ?

**Comment by gaby** [comment]
@nsarrazin this can be closed now
